DQM/SiStripMonitorCluster/src/SiStripMonitorCluster.cc

```diff
@@ -688,40 +706,115 @@ void SiStripMonitorCluster::analyze(const edm::Event& iEvent, const edm::EventSe
       // DetId and corresponding set of MEs
       ModMEs mod_single;
       if (Mod_On_) {
-	std::map<uint32_t, ModMEs >::iterator imodME = ModuleMEsMap.find(detid);
-	if (imodME != ModuleMEsMap.end()) {
-	  mod_single = imodME->second;
-	  found_module_me = true;
-	}
+        std::map<uint32_t, ModMEs >::iterator imodME = ModuleMEsMap.find(detid);
+        if (imodME != ModuleMEsMap.end()) {
+          mod_single = imodME->second;
+          found_module_me = true;
+        }
       } else found_module_me = false;
 
+
+
       edmNew::DetSetVector<SiStripCluster>::const_iterator isearch = cluster_detsetvektor->find(detid); // search  clusters of detid
+      edmNew::DetSet<SiStripCluster> cluster_detset = (*isearch);
+
+
+      //////////////////////////////////////////////////////////////
+
+      // Get all FED connections associated with given detID.
+      // All connections for a detid have same FED Id therefore one FEDID is associated with a given detID.
+      // Vector of constant FedChannelConnection objects to variable pointers.
+
+      // cout << "SiStripMonitorCluster::analyze >> =============== NEXT detid = " << detid << endl;
+      std::vector<const FedChannelConnection*> Fedconnections = SiStripDetCabling_->getConnections(detid);
+
+      for(unsigned int f=0; f<Fedconnections.size();f++){
+        if(f>0 && Fedconnections.at(f)!=NULL && Fedconnections.at(f-1)!=NULL){
```

Comment:
hi @Wilsker - please start the look at f=1, eliminate the unneeded checks on f and use [] instead of at.
(but in the end, this loop is useless, no?)